src/PrestaShopBundle/Entity/Store.php

```diff
@@ -0,0 +1,325 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints as PrestaShopAssert;
+use PrestaShopBundle\Entity\Repository\StoreRepository;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Table()
+ * @ORM\Entity(repositoryClass=StoreRepository::class)
+ */
+class Store
+{
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(name="id_store", type="integer", options={"unsigned":true})
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(name="id_country", type="integer", options={"unsigned"=true})
+     */
+    private $countryId;
+
+    /**
+     * @ORM\Column(name="id_state", type="integer", nullable=true, options={"unsigned"=true})
+     */
+    private $stateId;
+
+    /**
+     * @PrestaShopAssert\TypedRegex("post_code")
```

Comment:
An entity should always be valid: https://sarvendev.com/2018/01/entity-should-always-be-valid/
```suggestion
```